spyder/plugins/completion/providers/languageserver/widgets/status.py

```diff
@@ -113,37 +114,28 @@ def get_icon(self):
     @Slot()
     def update_status(self, lsp_language=None, status=None):
         """Update status message."""
-        editor_language = self.get_current_editor_language()
-
         # This case can only happen when switching files in the editor
         if lsp_language is None and status is None:
-            lsp_language = editor_language.lower()
-            if not self.plugin.clients.get(lsp_language, None):
+            lsp_language = self.current_language.lower()
+            if not self.provider.clients.get(lsp_language, None):
                 self.setVisible(False)
             else:
-                status = self.plugin.clients_statusbar.get(
+                status = self.provider.clients_statusbar.get(
                     lsp_language,
                     ClientStatus.STARTING
                 )
-                self.set_status(editor_language, status)
+                self.set_status(lsp_language, status)
                 self.setVisible(True)
             return
 
         # Don't update the status in case the editor and LSP languages
         # are different.
-        if editor_language.lower() != lsp_language:
+        if self.current_language.lower() != lsp_language:
             return
         else:
-            self.set_status(editor_language, status)
+            self.set_status(self.current_language, status)
             self.setVisible(True)
 
-    def get_current_editor_language(self):
-        """Get current LSP language."""
-        main = self.plugin.main
-        language = _('Unknown')
-
-        if main and main.editor:
-            codeeditor = main.editor.get_current_editor()
-            if codeeditor is not None:
-                language = codeeditor.language
-        return language
```

Comment:
I added this to handle the case where the language server is still not up and running, but you change to another file in the editor and come back here.

How are handling that now?